code/src/App.js

```diff
@@ -1,9 +1,58 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { PostedThought } from './components/PostedThought'
+import './App.css'
+import { NewThought } from './components/NewThought'
+
 
 export const App = () => {
+  const [postedThought, setPostedThought] = useState([])
+  const [newPostedThought, setNewPostedThought] = useState('')
+
+  useEffect(() => {
+    fetch('https://technigo-thoughts.herokuapp.com/', { method: 'GET' })
+      .then(res => res.json())
+      .then(json => setPostedThought(json))
+  }, [newPostedThought])
+
+  const handleFormSubmit = (newThought) => {
+    fetch('https://technigo-thoughts.herokuapp.com/', {
+      method: 'POST',
+      body: JSON.stringify({ message: newThought }),
+      headers: { 'Content-Type': 'application/json' }
+    })
+      .then(() => setNewPostedThought(newThought))
+      .catch(err => console.log("error:", err))
+
+  }
+
+  const onThoughtLiked = (likedThoughtId) => {
+    const updatedThoughts = postedThought.map((thought) => {
+      if (thought._id === likedThoughtId) {
+        thought.hearts += 1
+      }
+      return thought
+    })
+    setPostedThought(updatedThoughts)
+  }
+
   return (
     <div>
-      Find me in src/app.js!
+      <NewThought
+        onFormSubmit={handleFormSubmit}
+      />
+      {postedThought.map((thought) => (
+        <PostedThought
+          key={thought._id}
+          _id={thought._id}
```

Comment:
Just pointing it out if you don't know - you don't have to name this `_id`, you can name it `id` so it looks like the others.